### PR TITLE
[FIX] Can't copy or edit media's description

### PR DESCRIPTION
**Pull request by kresnaputra**
## Proposed changes

If message has attachment, I copy the text from message.attachments[0].description, otherwise I just copy from message.msg

## How to test or reproduce

- Edit messages text that have image in it:
1. Go to channel or DM.
2. Upload some image and add the caption
3. Edit the message.

- Copy messages text that have image in it:
1. Go to channel or DM.
2. Upload some image and add the caption
3. Copy the message.

## Types of changes

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] Improvement (non-breaking change which improves a current function)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Documentation update (if none of the other choices apply)

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/RocketChat/Rocket.Chat/blob/develop/.github/CONTRIBUTING.md#contributing-to-rocketchat) doc
- [x] I have signed the [CLA](https://cla-assistant.io/RocketChat/Rocket.Chat.ReactNative)
- [x] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works (if applicable)
- [ ] I have added necessary documentation (if applicable)
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: app/containers/MessageActions/index.js [hunk 1]
```diff
@@ -171,7 +171,7 @@ const MessageActions = React.memo(forwardRef(({
 
 	const handleCopy = async(message) => {
 		logEvent(events.ROOM_MSG_ACTION_COPY);
-		await Clipboard.setString(message.msg);
+		await Clipboard.setString(message.attachments.length > 0 ? message.attachments[0].description : message.msg);
```
Comment (diegolmello) [inline app/containers/MessageActions/index.js:174]: Can you simplify with something like `message?.attachments?.[0]?.description || message.msg`?

File: app/views/RoomView/index.js [hunk 2]
```diff
@@ -531,6 +531,9 @@ class RoomView extends React.Component {
 	}
 
 	onEditInit = (message) => {
+		if (message.attachments.length > 0) {
+			message._raw.msg = message.attachments[0].description;
```
Comment (diegolmello) [inline app/views/RoomView/index.js:535]: `message` on this context is an observable from WatermelonDB, so it shouldn't be changed directly (it's error prone).
Instead, you can pick only the properties needed from the object to do the edit.
Also, are you sure this is going to work?

**Review by diegolmello — CHANGES_REQUESTED** [review]

File: app/views/RoomView/index.js (hunk 2 above)
Comment (kresnaputra) [reply app/views/RoomView/index.js:535]: I tested it and it is work well, but like you said I need change the code to make sure the db is not change directly

**Review by diegolmello — COMMENTED** [review]
@kresnaputra Can you add details to test/reproduce this?

**Comment by kresnaputra** [comment]
@diegolmello updated

**Comment by diegolmello** [comment]
@kresnaputra Does it work for other types of attachments, like Video and Audio?

**Comment by kresnaputra** [comment]
@diegolmello yes, I already tried and it is work

**Review by diegolmello — APPROVED** [review]

**Pull request merged**